lib/wikidata.rb

```diff
@@ -19,10 +19,16 @@ def self.ids_from_pages(lang, titles)
         token_type: false,
       }
       response = client.action :query, page_args
-      redirected_from = Hash[(response.data['redirects'] || []).map { |h| [h['to'], h['from']] }]
-      response.data['pages'].select { |_k, v| v.key? 'pageprops' }.map do |_k, v|
-        [redirected_from[v['title']] || v['title'], v['pageprops']['wikibase_item']]
+      redirected_from = Hash[(response.data['redirects'] || []).map { |h| [h['from'], h['to']] }]
+      titles_and_ids = response.data['pages'].select { |_k, v| v.key? 'pageprops' }.map do |_k, v|
+        [v['title'], v['pageprops']['wikibase_item']]
       end
+      redirect_titles_and_ids = redirected_from.map do |k, v|
+        title_and_id = titles_and_ids.find { |a| a.include? v }
```

Comment:
This is odd on two fronts: 

* firstly, the `a.include?` is also at a very odd level of specificity, that disguises what you're actually testing. (i.e. that the first element of the Array == `v`).
* but secondly, you really shouldn't be calling `find` against an Array in a loop anyway — it would be much better to convert the original array to a Hash to check against.